pkgs/test/by-name/src/eval.rs

```diff
@@ -0,0 +1,114 @@
+use crate::structure;
+use crate::utils::ErrorWriter;
+
+use anyhow::Context;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::ffi::OsString;
+use std::io;
+use std::path::PathBuf;
+use std::process;
+use tempfile::NamedTempFile;
+
+/// Attribute set of this structure is returned by eval.nix
+#[derive(Deserialize)]
+struct AttributeInfo {
+    call_package_file: Option<PathBuf>,
+    is_derivation: bool,
+}
+
+const EXPR: &str = include_str!("eval.nix");
+
+pub fn check_values<W: io::Write>(
+    writer: &mut ErrorWriter<W>,
+    nixpkgs: &structure::Nixpkgs,
+    extra_nix_path: &OsStr,
+) -> anyhow::Result<()> {
+    let attrs_file = NamedTempFile::new().context("Failed to create a temporary file")?;
+    serde_json::to_writer(&attrs_file, &nixpkgs.package_names).context(format!(
+        "Failed to serialise the package names to the temporary path {}",
+        attrs_file.path().display()
+    ))?;
+
+    let mut nix_path = OsString::new();
+    nix_path.push(attrs_file.path());
```

Comment:
Please make it clear you are constructing a search path separated by `:`.

Reply:
I'm now getting around having to manually assemble the `NIX_PATH` by just passing entries to the CLI using `-I`.